RulesEngine/RuleProcessor.cs

```diff
@@ -164,24 +150,41 @@ public Issue[] Analyze(string text, LanguageInfo languageInfo)
                                     Rule = rule
                                 };
 
-                                //check at pattern level to avoid adding duplicates 
-                                if (_uniqueTagMatchesOnly && !UniqueTagsCheck(rule.Tags))
-                                    break;
+
+                                if (_uniqueTagMatchesOnly)
+                                {
+                                    if (!UniqueTagsCheck(newMatch.Rule.Tags)) //tag(s) previously seen
+                                    {
+                                        if (_stopAfterFirstPatternMatch) //recheck stop at pattern level also within same rule
+                                        {
+                                            passedConditions = false; //user performance option i.e. only wants to identify if tag is detected nothing more
+                                        }
+                                        else if (newMatch.Confidence > Confidence.Low) //user prefers highest confidence match over first match
+                                        {
+                                            passedConditions = BestMatch(matchList, newMatch); //; check all patterns in current rule
+
+                                            if (passedConditions)
+                                                passedConditions = BestMatch(resultsList, newMatch);//check all rules in permanent list                                         
+                                        }
+                                    }
+                                }
+
+                                if (passedConditions)
+                                {
+                                    matchList.Add(newMatch);
+                                }
 
                                 AddRuleTagHashes(rule.Tags);
```

Comment:
Should the tags be added if the match didn't pass conditions?

Reply:
It already passed see line 173 for being added as a tag the additional checks only apply to filtering for unique first/best settings.